File: demo/src/java/ast/KoKoLet.java
/**
 @author loriacarlos@gmail.com
 
*/
package kokoslan.ast;
import java.io.*;


public class KoKoLet implements KoKoAst{
   private KoKoAst id;
   public KoKoAst getId(){return this.id;}
   
   private KoKoAst expr;
   public KoKoAst getExpr(){return this.expr;}
   
   public KoKoLet(KoKoAst id, KoKoAst expr){
      this.id = id;
	  this.expr = expr;
   }
   public void genCode(PrintStream out){
      out.print("let ");
	  this.id.genCode(out);
	  out.print(" = ");
	  this.expr.genCode(out);
	  out.println();
   }
   public KoKoValue eval(KoKoContext ctx){
	   KoKoValue val = expr.eval(ctx);
	   ctx.assoc((KoKoId)id, val);
	   return val;
   }
}

File: KoKoslan/src/kotlin/ast/KoKoLet.kt
/** 
   Andrey Arguedas Espinoza
   Daniela Armas Sanchez
   Michael Chen Wang
   Kimberly Olivas Delgado
*/

package kokoslan.kt.ast
import java.io.*
import kokoslan.kt.eval.*
import kokoslan.kt.exception.*

class KoKoLet(private var id : KoKoAst, private var expr : KoKoAst) : KoKoAst {

   fun getId() : KoKoAst {return this.id}
   fun getExpr() : KoKoAst {return this.expr}
   
   override fun genCode(Out : PrintStream) : Unit{
      Out.print("let ")
	  this.id.genCode(Out)
	  Out.print(" = ")
	  this.expr.genCode(Out)
	  Out.println()
   }
   
   override fun eval(ctx : KoKoContext) : KoKoValue?{
	   val value = expr.eval(ctx)
	   ctx.assoc(id as KoKoId, value) //Parsing?
	   return value;
   }
}